rumqttd/src/protocol/v4/connack.rs

```diff
@@ -51,7 +51,7 @@ fn connect_code(return_code: ConnectReturnCode) -> u8 {
     match return_code {
         ConnectReturnCode::Success => 0,
         ConnectReturnCode::RefusedProtocolVersion => 1,
-        ConnectReturnCode::BadClientId => 2,
+        ConnectReturnCode::ClientIdentifierNotValid => 2,
```

Comment:
unrelated to this PR, but in v5 this connect reason codes are not valid, see:

https://docs.oasis-open.org/mqtt/mqtt/v5.0/os/mqtt-v5.0-os.html#_Toc3901079
https://docs.oasis-open.org/mqtt/mqtt/v5.0/os/mqtt-v5.0-os.html#_Toc3901079

can you open an issue for this

Reply:
wdym by they are wrong? also the quoted line only affects v4.